### PR TITLE
Configure Nunjucks from `lib/nunjucks`, consolidate slugs

### PR DESCRIPTION
**Pull request by colinrotherham**
This PR configures Nunjucks filters, globals and options from a single place

We previously had three ways to generate a slug, which was hard to follow:

1. [`slugger` npm package](https://www.npmjs.com/package/slugger)
2. [`kebabCase` Nunjucks filter](https://github.com/alphagov/govuk-design-system/blob/6cf12f37f28c0b9575d48e9e75af29669887e4f1/lib/metalsmith.js#L64-L66)
3. [`options.slug` for macro options](https://github.com/alphagov/govuk-design-system/blob/6cf12f37f28c0b9575d48e9e75af29669887e4f1/lib/get-macro-options/index.js#L17-L18)

This caused differences in output HTML, for example `formGroup` versus `form-group`

I've already consolidated 1) and 2) in https://github.com/alphagov/govuk-design-system/pull/2927, but now have a [single `slugify` filter to match GOV.UK Frontend](https://github.com/alphagov/govuk-frontend/pull/3963)

### CODE REVIEW

**Comment by colinrotherham** [comment]
Good to apply the same improvements to GOV.UK Frontend too, so this one is related:

* https://github.com/alphagov/govuk-frontend/pull/4092

**Comment by colinrotherham** [comment]
@romaricpascal Yeah some URLs have changed

Previously we could easily create a `#test-errorSummary` fragment for a non-matching `id="test-error-summary`

Some of these were [fixed a while back](https://github.com/alphagov/govuk-design-system/commit/32dec13ae003e6ec12890e4962c09d9f33ef51ae) but it wasn't rolled out to all slugs

**Comment by colinrotherham** [comment]
Diff for this particular PR picks up on a few remaining slug with symbols (that should be hyphens)

```patch
--- A/deploy/public/components/summary-list/index.html
+++ B/deploy/public/components/summary-list/index.html
@@ -1089,12 +1089,12 @@
                     <td class="govuk-table__cell">array</td>
                     <td class="govuk-table__cell">
                       Array of action item objects.
+                        See <a href="#options-summary-list-example--actions-items">actions.<wbr>items</a>.
-                        See <a href="#options-summary-list-example--actions.items">actions.<wbr>items</a>.
 </td>
                   </tr>
               </tbody>
             </table>
+            <table class="govuk-table app-options__table" id="options-summary-list-example--actions-items">
-            <table class="govuk-table app-options__table" id="options-summary-list-example--actions.items">
               <caption class="govuk-table__caption govuk-heading-m ">Options for actions.<wbr>items</caption>
               <thead class="govuk-table__head">
                 <tr class="govuk-table__row">
@@ -1609,12 +1609,12 @@
                     <td class="govuk-table__cell">array</td>
                     <td class="govuk-table__cell">
                       Array of action item objects.
+                        See <a href="#options-without-actions-summary-list-example--actions-items">actions.<wbr>items</a>.
-                        See <a href="#options-without-actions-summary-list-example--actions.items">actions.<wbr>items</a>.
 </td>
                   </tr>
               </tbody>
             </table>
+            <table class="govuk-table app-options__table" id="options-without-actions-summary-list-example--actions-items">
-            <table class="govuk-table app-options__table" id="options-without-actions-summary-list-example--actions.items">
               <caption class="govuk-table__caption govuk-heading-m ">Options for actions.<wbr>items</caption>
               <thead class="govuk-table__head">
                 <tr class="govuk-table__row">
@@ -2103,12 +2103,12 @@
                     <td class="govuk-table__cell">array</td>
                     <td class="govuk-table__cell">
                       Array of action item objects.
+                        See <a href="#options-summary-list-second-example--actions-items">actions.<wbr>items</a>.
-                        See <a href="#options-summary-list-second-example--actions.items">actions.<wbr>items</a>.
 </td>
                   </tr>
               </tbody>
             </table>
+            <table class="govuk-table app-options__table" id="options-summary-list-second-example--actions-items">
-            <table class="govuk-table app-options__table" id="options-summary-list-second-example--actions.items">
               <caption class="govuk-table__caption govuk-heading-m ">Options for actions.<wbr>items</caption>
               <thead class="govuk-table__head">
                 <tr class="govuk-table__row">
@@ -2624,12 +2624,12 @@
                     <td class="govuk-table__cell">array</td>
                     <td class="govuk-table__cell">
                       Array of action item objects.
+                        See <a href="#options-summary-list-with-a-mix-of-rows-with-and-without-actions-summary-list-example--actions-items">actions.<wbr>items</a>.
-                        See <a href="#options-summary-list-with-a-mix-of-rows-with-and-without-actions-summary-list-example--actions.items">actions.<wbr>items</a>.
 </td>
                   </tr>
               </tbody>
             </table>
+            <table class="govuk-table app-options__table" id="options-summary-list-with-a-mix-of-rows-with-and-without-actions-summary-list-example--actions-items">
-            <table class="govuk-table app-options__table" id="options-summary-list-with-a-mix-of-rows-with-and-without-actions-summary-list-example--actions.items">
               <caption class="govuk-table__caption govuk-heading-m ">Options for actions.<wbr>items</caption>
               <thead class="govuk-table__head">
                 <tr class="govuk-table__row">
@@ -3124,12 +3124,12 @@
                     <td class="govuk-table__cell">array</td>
                     <td class="govuk-table__cell">
                       Array of action item objects.
+                        See <a href="#options-without-borders-summary-list-example--actions-items">actions.<wbr>items</a>.
-                        See <a href="#options-without-borders-summary-list-example--actions.items">actions.<wbr>items</a>.
 </td>
                   </tr>
               </tbody>
             </table>
+            <table class="govuk-table app-options__table" id="options-without-borders-summary-list-example--actions-items">
-            <table class="govuk-table app-options__table" id="options-without-borders-summary-list-example--actions.items">
               <caption class="govuk-table__caption govuk-heading-m ">Options for actions.<wbr>items</caption>
               <thead class="govuk-table__head">
                 <tr class="govuk-table__row">
@@ -3744,12 +3744,12 @@
                     <td class="govuk-table__cell">array</td>
                     <td class="govuk-table__cell">
                       Array of action item objects.
+                        See <a href="#options-summary-in-a-card-with-a-title-summary-list-example--actions-items">actions.<wbr>items</a>.
-                        See <a href="#options-summary-in-a-card-with-a-title-summary-list-example--actions.items">actions.<wbr>items</a>.
 </td>
                   </tr>
               </tbody>
             </table>
+            <table class="govuk-table app-options__table" id="options-summary-in-a-card-with-a-title-summary-list-example--actions-items">
-            <table class="govuk-table app-options__table" id="options-summary-in-a-card-with-a-title-summary-list-example--actions.items">
               <caption class="govuk-table__caption govuk-heading-m ">Options for actions.<wbr>items</caption>
               <thead class="govuk-table__head">
                 <tr class="govuk-table__row">
@@ -4443,12 +4443,12 @@
                     <td class="govuk-table__cell">array</td>
                     <td class="govuk-table__cell">
                       Array of action item objects.
+                        See <a href="#options-summary-in-a-card-with-a-title-and-actions-summary-list-example--actions-items">actions.<wbr>items</a>.
-                        See <a href="#options-summary-in-a-card-with-a-title-and-actions-summary-list-example--actions.items">actions.<wbr>items</a>.
 </td>
                   </tr>
               </tbody>
             </table>
+            <table class="govuk-table app-options__table" id="options-summary-in-a-card-with-a-title-and-actions-summary-list-example--actions-items">
-            <table class="govuk-table app-options__table" id="options-summary-in-a-card-with-a-title-and-actions-summary-list-example--actions.items">
               <caption class="govuk-table__caption govuk-heading-m ">Options for actions.<wbr>items</caption>
               <thead class="govuk-table__head">
                 <tr class="govuk-table__row">
```

**Review by romaricpascal — APPROVED** [review]
That's not many changes 🥳 And these links are for nunjucks options that are burried into hidden tabs, so we can merge without worries ⛵
